src/test/java/com/google/cloud/teleport/templates/common/SplunkConvertersTest.java

```diff
@@ -309,6 +313,110 @@ public void testFailsafeStringToSplunkEventValidIndex() {
     pipeline.run();
   }
 
+  /** Test successful conversion of JSON messages with user provided index 'fields'. */
```

Comment:
As several tests are repeating same logic except wit change in input and output the logic can be moved to different method (e.g: matchesSplunkEvent)
```
@Category(NeedsRunner.class)
  private void matchesSplunkEvent(FailsafeElement<String, String> input,SplunkEvent expectedSplunkEvent) {
    pipeline.getCoderRegistry().registerCoderForClass(SplunkEvent.class, SplunkEventCoder.of());

    PCollectionTuple tuple =
            pipeline
                    .apply(
                            Create.of(input)
                                    .withCoder(FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())))
                    .apply(
                            SplunkConverters.failsafeStringToSplunkEvent(
                                    SPLUNK_EVENT_OUT, SPLUNK_EVENT_DEADLETTER_OUT));

    PAssert.that(tuple.get(SPLUNK_EVENT_DEADLETTER_OUT)).empty();
    PAssert.that(tuple.get(SPLUNK_EVENT_OUT))
            .containsInAnyOrder(expectedSplunkEvent);

    pipeline.run();
  }
```
Now tests can be simplified to call the method with just input & expectedOutput. For e.g:
```
@Test
  public void testFailsafeStringToSplunkEvent_withValidFields() {

    FailsafeElement<String, String> input =
        FailsafeElement.of(
            "",
            "{\n"
                + "\t\"name\": \"Jim\",\n"
                + "\t\"_metadata\": {\"fields\":{\"test-key\":\"test-value\"}}\n"
                + "}");

    SplunkEvent expectedSplunkEvent = SplunkEvent.newBuilder()
                                    .withEvent("{\"name\":\"Jim\"}")
                                    .withFields(GSON.fromJson("{\"test-key\":\"test-value\"}", JsonObject.class))
                                    .build();

    matchesSplunkEvent(input, expectedSplunkEvent);
  }
 ```

Reply:
Moved the test pipeline to `matchesSplunkEvent()`